app/assets/javascripts/components/generic_object/custom-image-component.js

```diff
@@ -0,0 +1,89 @@
+ManageIQ.angular.app.component('customImageComponent', {
+  bindings: {
+    picture: '=',
+    newRecord: '<',
+    angularForm: '<',
+    pictureUrlPath: '@',
+    pictureUploaded: '=',
+    pictureRemove: '=',
+    pictureReset: '<',
+  },
+  controllerAs: 'vm',
+  controller: customImageComponentController,
+  templateUrl: '/static/generic_object/custom-image-component.html.haml',
+});
+
+customImageComponentController.$inject = ['$timeout'];
+
+function customImageComponentController($timeout) {
+  var vm = this;
+
+  vm.$onInit = function() {
+    vm.imageUploadStatus = "";
+    vm.changeImage = false;
+  };
+
+  vm.$onChanges = function() {
+    vm.changeImage = false;
+    restoreOriginalStatus();
+  };
+
+  vm.uploadClicked = function() {
+    var imageFile;
+
+    if (angular.element('#generic_object_definition_image_file')[0].files.length === 0) {
+      return;
+    }
+
+    imageFile = angular.element('#generic_object_definition_image_file')[0].files[0];
+
+    var reader = new FileReader();
+    vm.imageUploadStatus = "";
+
+    if (imageFile.type === 'image/png') {
+      vm.picture.extension = 'png';
+    } else if (imageFile.type === 'image/jpg') {
+      vm.picture.extension = 'jpg';
+    } else if (imageFile.type === 'image/jpeg') {
+      vm.picture.extension = 'jpeg';
+    } else {
+      vm.angularForm.generic_object_definition_image_file_status.$setValidity("incompatibleFileType", false);
+      vm.imageUploadStatus = __("Incompatible image type");
+      vm.pictureUploaded = true;
+      return;
+    }
+
+    reader.onload = function(event) {
+      vm.picture.content = btoa(event.target.result);
+
+      $timeout(function() {
+        vm.angularForm.generic_object_definition_image_file_status.$setValidity("incompatibleFileType", true);
+        vm.imageUploadStatus = __("Image upload complete");
```

Comment:
Would a different message make more sense here given this does not actually upload the image yet?

I mean, if I see "Image upload complete" I will think the image is already on the server. (And will wonder why submitting the form takes so long.)

(I guess there's no support for uploading the image in the background yet, right?)

Reply:
"Image upload complete" is just a helpful acknowledgement to the user that the chosen file is now ready to be saved.
LMK if you have other wording suggestions for that message.

> I guess there's no support for uploading the image in the background yet, right?

No, it all happens in the main thread.
In my testing so far, I haven't come across a situation where the file loading is like a really lengthy, time-consuming operation  -- it has been almost instantaneous in my tests.

Reply:
Sounds good. Will change that text.

Reply:
On second thoughts, saying "...submit the form" sounds a bit techy, let's try and keep it very generic. 
How about "Image is ready to be uploaded"?

Reply:
Addressed in fb22a12

Reply:
Better, thanks! :)